PHP/PHP Source.sublime-syntax

```diff
@@ -2016,18 +2569,214 @@ contexts:
+    - match: (?=\${{identifier_start}})
+      push: regexp-interpolation-literal-variable
+
+  regexp-interpolation-braced-expression:
+    - clear_scopes: 2
+    - meta_include_prototype: false
+    - meta_scope: source.regexp.php meta.interpolation.php
+    - include: interpolation-braced-expression
+
+  regexp-interpolation-braced-variable:
+    - clear_scopes: 2
+    - meta_include_prototype: false
+    - meta_scope: source.regexp.php meta.interpolation.php meta.variable.php
+    - include: interpolation-braced-variable
+
+  regexp-interpolation-literal-variable:
+    - clear_scopes: 2
+    - meta_include_prototype: false
+    - meta_scope: source.regexp.php meta.interpolation.php
+    - include: interpolation-literal-variable
+
+###[ OPERATORS ]##############################################################
+
+  operators:
+    - include: variadic-operators
+    - match: <\?(?i:php\b|(?!php)ph?|=)?
+      scope: punctuation.section.embedded.begin.php
+    - match: =>
+      scope: punctuation.separator.key-value.php
+    - match: (?:\+|-|\*|/|%|&|\||\^|>>|<<|\.|\?\?)=
+      scope: keyword.operator.assignment.augmented.php
+    - match: (?:(=)(&))|(&(?=[[:alpha:]$_]))
+      captures:
+        1: keyword.operator.assignment.php
+        2: keyword.operator.reference.php
+        3: keyword.operator.reference.php
+    - match: \|\||&&
+      scope: keyword.operator.logical.php
+    - match: <<|>>|~|\^|&|\|
+      scope: keyword.operator.bitwise.php
+    - match: ===|!==|<=>|==|!=|<=|>=|<>|<|>
+      scope: keyword.operator.comparison.php
+    - match: \?\?
+      scope: keyword.operator.null-coalescing.php
+    # single char operators
+    - match: '[-+*/%]'
+      scope: keyword.operator.arithmetic.php
+    - match: =
+      scope: keyword.operator.assignment.php
+    - match: \@
+      scope: keyword.operator.error-control.php
+    - match: \!
+      scope: keyword.operator.logical.php
+    - match: \.
+      scope: keyword.operator.concatenation.php
+    - match: \?(?!>)
+      scope: keyword.operator.ternary.php
+      push: ternary-expression
+    - match: ':(?!:)'
+      scope: punctuation.separator.colon.php
+    - include: punctuation-separators
+    # word-like operators
+    - match: '{{logical_operators}}'
+      scope: keyword.operator.logical.php
+    - match: (?i:instanceof)\b
+      scope: keyword.other.instanceof.php
+      push: instanceof-operand
+
+  dereference-operators:
+    - match: \$(?=\$)
+      scope: keyword.operator.dereference.php
+    - match: \$
+      scope: punctuation.definition.variable.php
+
+  instanceof-operand:
+    - include: binding
+    - include: class-name
+    - include: attributes
+    - include: comments
+    - include: else-pop
+
+  ternary-expression:
+    - match: ':(?!:)'
+      scope: keyword.operator.ternary.php
+      pop: 1
+    - include: nested-expressions
+
+  variadic-operators:
+    - match: \.\.\.
+      scope: keyword.operator.variadic.php
```

Comment:
Not sure how to do this or if you already did, but this syntax is also used for first-class callables in PHP 8.1+, so it can appear in other contexts now: https://www.php.net/manual/en/functions.first_class_callable_syntax.php.

Maybe that should just be added as an additional scope name?

Reply:
It's not explicitly implemented but those constructs seem properly handled, except `...` being scoped `variadic`.

`->myMethod` is scoped as ordinary function-call, not sure it should receive any special treatment.